This is VERY important, so you know.

Without it, atapi CDR/CDRW completely doesn't work.

------- Forwarded Message

Forwarded: Tue, 09 Nov 1999 12:25:51 -0700
Forwarded: "turan "
Forwarded: Wed, 03 Nov 1999 00:23:12 -0700
Forwarded: "turan "
Return-Path: csapuntz@adsl-63-192-219-28.dsl.snfc21.pacbell.net
Received: from mta2.snfc21.pbi.net (mta2.snfc21.pbi.net [206.13.28.123])
	by cvs.openbsd.org (8.9.3/8.9.1) with ESMTP id AAA02243;
	Wed, 3 Nov 1999 00:14:45 -0700 (MST)
Received: from adsl-63-192-219-28.dsl.snfc21.pacbell.net ([63.192.219.28])
 by mta2.snfc21.pbi.net (Sun Internet Mail Server sims.3.5.1999.09.16.21.57.p8)
 with ESMTP id <0FKM00E791FEAW@mta2.snfc21.pbi.net>; Tue,
 2 Nov 1999 23:14:02 -0800 (PST)
Received: (from csapuntz@localhost)
 by adsl-63-192-219-28.dsl.snfc21.pacbell.net (8.9.3/8.9.1) id XAA32134; Tue,
 02 Nov 1999 23:14:21 -0800 (PST)
Date: Tue, 02 Nov 1999 23:14:21 -0800 (PST)
From: Constantine Sapuntzakis <csapuntz@stanford.edu>
Subject: additional patch for cdrecord
To: deraadt@cvs.openbsd.org, niklas@cvs.openbsd.org
Message-id: <199911030714.XAA32134@adsl-63-192-219-28.dsl.snfc21.pacbell.net>
MIME-version: 1.0 (generated by tm-edit 7.108)
Content-type: text/plain; charset=US-ASCII


I called this patch-bb.

- -Costa

- --- cdrecord/scsi-bsd.c-	Fri Oct  9 04:21:39 1998
+++ cdrecord/scsi-bsd.c 	Tue Nov  2 22:49:29 1999
@@ -56,7 +56,13 @@
 #define	SADDR_TARGET(a)	(SADDR_ISSCSI(a)?(a).addr.scsi.target:(a).addr.atapi.atbus*2+(a).addr.atapi.drive)
 #define	SADDR_LUN(a)	(SADDR_ISSCSI(a)?(a).addr.scsi.lun:0)
 #else
+
+#if     defined(__OpenBSD__) && defined(TYPE_ATAPI)
+#define MAYBE_ATAPI
+#define	SADDR_ISSCSI(a)	((a).type == TYPE_SCSI)
+#else
 #define	SADDR_ISSCSI(a)	(1)
+#endif  /* OpenBSD */
 
 #define	SADDR_BUS(a)	(a).scbus
 #define	SADDR_TARGET(a)	(a).target


------- End of Forwarded Message
